sound/hda/intel-dsp-config.c

```diff
@@ -186,6 +194,10 @@ static const struct config_entry config_table[] = {
 			{}
 		}
 	},
+	{
+		.flags = FLAG_SOF | FLAG_SOF_ONLY_IF_SOUNDWIRE,
+		.device = 0xa348,
+	},
 	{
 		.flags = FLAG_SOF | FLAG_SOF_ONLY_IF_DMIC,
 		.device = 0xa348,
```

Comment:
Hmm, this seems a bit odd to have two entries for many devices (like 0xa348 here). 

FLAG_SOF_ONLY_IF_NON_HDA_CODECS

... if anything that legacy driver can't handle (dmic, soundwire or I2s codecs), use SOF, but only then.

Reply:
it can't be NON_HDA_CODECS since it would imply I2S. I added DMIC_OR_SOUNDWIRE to avoid ambiguities

Reply:
fixed